src/views/DatasetManager/components/DatasetList/DatasetList.js

```diff
@@ -0,0 +1,138 @@
+import { DeleteForever as DeleteForeverIcon, Refresh as RefreshIcon } from '@mui/icons-material';
+import { DontAskAgainDialog } from '@cosmotech/ui';
+import { CreateDatasetButton } from '../../../../components';
+import { DatasetsUtils } from '../../../../utils';
+import { useTranslation } from 'react-i18next';
+import { useDatasetList } from './DatasetListHook';
+import { TwoActionsDialogService } from '../../../../services/twoActionsDialog/twoActionsDialogService';
+
+export const DatasetList = () => {
+  const { t } = useTranslation();
+  const [isRefreshConfirmationDialogOpen, setIsRefreshConfirmationDialogOpen] = useState(false);
+  const { sortedDatasetList, currentDataset, selectDataset } = useDatasetList();
+  const refreshDialogLabels = {
+    title: t('commoncomponents.datasetmanager.dialogs.refresh.title', 'Overwrite data?'),
+    body: t(
+      'commoncomponents.datasetmanager.dialogs.refresh.body',
+      'Your data will be lost, replaced with the imported one.'
+    ),
+    cancel: t('commoncomponents.datasetmanager.dialogs.cancel', 'Cancel'),
+    confirm: t('commoncomponents.datasetmanager.dialogs.refresh.overwriteButton', 'Overwrite'),
+    checkbox: t('commoncomponents.datasetmanager.dialogs.refresh.checkbox', 'Do not ask me again'),
+  };
+  const askConfirmationToDeleteDialog = useCallback(
+    async (event, datasetId) => {
+      event.stopPropagation();
+      const initialDatasetName = sortedDatasetList.find((dataset) => dataset.id === datasetId)?.name;
+      const datasetsToDelete = DatasetsUtils.getDatasetTree(datasetId, sortedDatasetList);
+      const dialogProps = {
+        id: 'delete-dataset',
+        component: 'div',
+        labels: {
+          title: t('commoncomponents.datasetmanager.dialogs.delete.title', 'Delete dataset'),
+          body: (
+            <>
+              {t(
+                'commoncomponents.datasetmanager.dialogs.delete.body',
+                'Do you really want to delete {{datasetName}}? This action is irreversible.',
+                { datasetName: initialDatasetName }
+              )}
+              <p>
+                {datasetsToDelete.length > 0 &&
+                  t(
+                    'commoncomponents.datasetmanager.dialogs.delete.bodyWithChildren',
+                    'The following children datasets will also be permanently removed:'
+                  )}
+              </p>
+              <List>
+                {datasetsToDelete.map((dataset) => (
+                  <ListItem
+                    key={dataset}
+                    sx={{ display: 'list-item', listStyleType: 'disc', listStylePosition: 'inside' }}
+                    dense
+                  >
+                    {dataset}
+                  </ListItem>
+                ))}
+              </List>
+            </>
+          ),
+          button1: t('commoncomponents.datasetmanager.dialogs.cancel', 'Cancel'),
+          button2: t('commoncomponents.datasetmanager.dialogs.delete.confirmButton'),
```

Comment:
A default value is missing here in the translation function

Reply:
Nice catch, thanks ! :+1: